### ISSUE TITLE
Is there a way to add custom properties on image import? (possible bug)

### ISSUE THREAD

**Issue by n0guest** [lxc/lxd#3803]
# Required information

 * Distribution: Ubuntu
 * Distribution version: 16.04/xenial
 * The output of "lxc info" or if that fails:

```
config:
  core.https_address: '[::]:8443'
  core.trust_password: true
api_extensions:
- storage_zfs_remove_snapshots
- container_host_shutdown_timeout
- container_syscall_filtering
- auth_pki
- container_last_used_at
- etag
- patch
- usb_devices
- https_allowed_credentials
- image_compression_algorithm
- directory_manipulation
- container_cpu_time
- storage_zfs_use_refquota
- storage_lvm_mount_options
- network
- profile_usedby
- container_push
- container_exec_recording
- certificate_update
- container_exec_signal_handling
- gpu_devices
- container_image_properties
- migration_progress
- id_map
- network_firewall_filtering
- network_routes
- storage
- file_delete
- file_append
- network_dhcp_expiry
- storage_lvm_vg_rename
- storage_lvm_thinpool_rename
- network_vlan
- image_create_aliases
- container_stateless_copy
- container_only_migration
- storage_zfs_clone_copy
- unix_device_rename
- storage_lvm_use_thinpool
- storage_rsync_bwlimit
- network_vxlan_interface
- storage_btrfs_mount_options
- entity_description
- image_force_refresh
- storage_lvm_lv_resizing
- id_map_base
- file_symlinks
- container_push_target
- network_vlan_physical
- storage_images_delete
- container_edit_metadata
- container_snapshot_stateful_migration
- storage_driver_ceph
- storage_ceph_user_name
- resource_limits
- storage_volatile_initial_source
api_status: stable
api_version: "1.0"
auth: trusted
public: false
environment:
  addresses:
  - <DELETED>
  architectures:
  - x86_64
  - i686
  certificate: |
     <DELETED>
  certificate_fingerprint: <DELETED>
  driver: lxc
  driver_version: 2.0.8
  kernel: Linux
  kernel_architecture: x86_64
  kernel_version: 4.4.0-83-generic
  server: lxd
  server_pid: 28627
  server_version: "2.17"
  storage: zfs
  storage_version: 0.6.5.6-0ubuntu16
```

# Issue description

I'm trying to import an image from tar.gz file with adding custom properties. But it always ends like this:

```
# lxc image import grafana.tar.gz --alias='grafana-tpl' user.field=123
panic: assignment to entry in nil map

goroutine 1 [running]:
panic(0x8645e0, 0xc8201487b0)
        /usr/lib/go-1.6/src/runtime/panic.go:481 +0x3e6
main.(*imageCmd).run(0xc82001d900, 0xc820114180, 0xc8201145c0, 0x3, 0x4, 0x0, 0x0)
        /build/lxd-XQb6fr/lxd-2.17/obj-x86_64-linux-gnu/src/github.com/lxc/lxd/lxc/image.go:689 +0x7136
main.run(0x0, 0x0)
        /build/lxd-XQb6fr/lxd-2.17/obj-x86_64-linux-gnu/src/github.com/lxc/lxd/lxc/main.go:167 +0x110d
main.main()
        /build/lxd-XQb6fr/lxd-2.17/obj-x86_64-linux-gnu/src/github.com/lxc/lxd/lxc/main.go:28 +0x5e
```

Looks like it's related to properties parsing (https://github.com/lxc/lxd/blob/lxd-2.17/lxc/image.go#L689):

```
		// Handle properties
		for _, entry := range properties {
			fields := strings.SplitN(entry, "=", 2)
			if len(fields) < 2 {
				return fmt.Errorf(i18n.G("Bad property: %s"), entry)
			}

			image.Properties[strings.TrimSpace(fields[0])] = strings.TrimSpace(fields[1])
		}
```

Though, without "user.field=123" it imports just fine. Am I missing anything or it's a bug?

Thanks.

**Comment by stgraber** [comment]
Looks like a bug, I'll take a look.

**Comment by stgraber** [comment]
Reproduced the crash here.

**Comment by stgraber** [comment]
got a fix

**Comment by n0guest** [comment]
> got a fix

Thanks. As I understand, this fix will be included in 2.18 (~19 Sep or later)?

**Comment by stgraber** [comment]
That's correct
